InvenTree/part/api.py

```diff
@@ -85,6 +85,8 @@ def filter_queryset(self, queryset):
 
         cascade = str2bool(params.get('cascade', False))
 
+        depth = params.get("depth", None)
```

Comment:
I would suggest that you try parsing `depth` to an int here, and set it to None if the integer coerce fails. 

- Won't throw an error if you pass something like `?depth=yes`
- You don't have to cast to int multiple times later on.